### PR TITLE
Add authorization for deposits and withdrawals

### PR DESCRIPTION
**Pull request by gcomte**

### CODE REVIEW

File: examples/js/tests/authorization.ts [hunk 1]
```diff
@@ -417,10 +491,14 @@ function restAssertAuthenticationNotSatisfactory(error) {
 }
 
 async function main() {
+  // disable logging
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  console.log = () => {};
```
Comment (gobeli) [inline examples/js/tests/authorization.ts:496]: That one seems a bit scary :grimacing:
We can leave it like that for now, but I would suggest having a central logger which can filter out logs based on a LOG_LEVEL env

File: market_test_output.txt [hunk 2]
```diff
@@ -0,0 +1,256 @@
+{"type":"BalanceMessage","value":{"timestamp":1634971683.0,"user_id":"b7c65c7f-7c73-4e3b-a450-2816df872929","business_id":0,"asset":"USDT","business":"deposit","market_price":"0","change":"1000000","balance":"1000000","balance_available":"1000000","balance_frozen":"0","detail":"{\"id\":0}"}}
```
Comment (gobeli) [inline market_test_output.txt]: Is this file needed?

**Comment by gobeli** [comment]
Great stuff :partying_face:

**Review by gobeli — APPROVED** [review]

File: market_test_output.txt (hunk 2 above)
Comment (gcomte) [reply market_test_output.txt]: Not it's not :-) I removed it.

File: examples/js/tests/authorization.ts (hunk 1 above)
Comment (gcomte) [reply examples/js/tests/authorization.ts:496]: Agreed! Let's leave it like this for now though, since we're still in the prototyping phase. I've created [an issue](https://github.com/gcomte/dingir-exchange/issues/27) for us to keep track of it.

**Comment by gcomte** [comment]
Problems resolved, I'll go ahead and merge it.
